Integrate Blog page with Netlify CMS data

mecies opened the pull request:
### Description

This PR covers:
- Update BlogPost component with onClick
- Update BlogPostDialog component with action handlers
- Refactor repetitive logic to determine the components type to a reusable hook
- Fetch data to blog-post page from CMS and add a missing image field in information structure
- Update stories for changed components

Closes #68  

### Preview
<img width="1372" alt="Screenshot 2020-09-09 at 22 57 23" src="https://user-images.githubusercontent.com/39158460/92654311-f6351180-f2ef-11ea-8eaa-0794ca6684ff.png">
<img width="751" alt="Screenshot 2020-09-09 at 22 57 31" src="https://user-images.githubusercontent.com/39158460/92654318-f7fed500-f2ef-11ea-868e-da431f4abec2.png">
<img width="831" alt="Screenshot 2020-09-09 at 22 57 35" src="https://user-images.githubusercontent.com/39158460/92654326-f9c89880-f2ef-11ea-84fa-deb59b769be2.png">

mecies committed:
Read blogpage data from content file

mecies committed:
Refactor determining component type to helper hook

mecies committed:
Add blogpost dialog with content

mecies committed:
Update coponent stories

mecies committed:
Improve component type hook

KonradSzwarc commented on src/pages/blog-page/index.tsx:
```diff
@@ -121,21 +127,66 @@ const useStyles = makeStyles((theme) => ({
-              {[...Array(4)].map((_, index) => (
-                <BlogPost key={index} className={classes.blogPost} {...blogData} />
+              {blogData.blog_post.map((blogPost, index) => (
+                <div key={index}>
```
Don't use the index as key

mecies replied:
Good catch, will replace in all pages aswell

KonradSzwarc commented on src/hooks/useComponentType.ts:
```diff
@@ -0,0 +1,10 @@
+  const isTablet = useMediaQuery(theme.breakpoints.down('md'));
+  const isDesktop = !isMobile && !isTablet;
+  const componentType: 'mobile' | 'tablet' | 'desktop' = isMobile ? 'mobile' : isTablet ? 'tablet' : 'desktop';
+  return { componentType, isMobile, isTablet, isDesktop };
```
Do we need to return `isMobile`, `isTablet`, and `isDasktop` if we can use `componentType` to determine the size

mecies replied:
I think that it is more elegant to return them.
I changed it as such in the last commit, without it everyone would have to do the check `componentType === "mobile"` etc and I think that this way is a better solution.

KonradSzwarc commented on src/components/BlogPost/BlogPostDialogTablet.tsx:
```diff
@@ -37,24 +37,24 @@ const useStyles = makeStyles((theme) => ({
           </Box>
           <Box mb={2.5}>
-            <IconButton {...props} size={'small'}>
+            <IconButton onClick={props.handlePrev} size={'small'}>
```
`size="small"`

mecies replied:
Good catch :D Will replace project-wide

mecies committed:
Cr fixes

Make keys unique for rendered lists and replace passing 'small' prop application-wide
